txscript/standard_test.go

```diff
@@ -1013,3 +1019,89 @@ func TestStringifyClass(t *testing.T) {
 		}
 	}
 }
+
+// TestNullDataScript tests whether NullDataScript returns a valid script.
+func TestNullDataScript(t *testing.T) {
+	tiny, _ := hex.DecodeString("01")
+	short, _ := hex.DecodeString("0102030405060708090a0b0c0d0e0f1" +
+		"01112131415161718")
+	shortExp, _ := hex.DecodeString("6a180102030405060708090a0b0c0d0e0f1" +
+		"01112131415161718")
+	justRight, _ := hex.DecodeString("000102030405060708090a0b0c0d0e0f" +
+		"101112131415161718191a1b1c1d1e1f20212223242526272829" +
+		"2a2b2c2d2e2f303132333435363738393a3b3c3d3e3f40414243" +
+		"4445464748494a4b4c4d4e4f")
+	tooLong, _ := hex.DecodeString("000102030405060708090a0b0c0d0e0f" +
+		"101112131415161718191a1b1c1d1e1f20212223242526272829" +
+		"2a2b2c2d2e2f303132333435363738393a3b3c3d3e3f40414243" +
+		"4445464748494a4b4c4d4e4f50")
+
+	tests := []struct {
+		name     string
+		data     []byte
+		valid    bool
+		expected []byte
+	}{
+		{
+			name:     "small data",
+			data:     tiny,
+			valid:    true,
+			expected: []byte{0x6a, 0x51},
+		},
+		{
+			name:     "data of size before OP_PUSHDATA1 is needed",
+			data:     short,
+			valid:    true,
+			expected: shortExp,
+		},
+		{
+			name:  "just right",
+			data:  justRight,
+			valid: true,
+			expected: append([]byte{
+				txscript.OP_RETURN, txscript.OP_PUSHDATA1, 80},
+				justRight...),
+		},
+		{
+			name:     "too big",
+			data:     tooLong,
+			valid:    false,
+			expected: nil,
+		},
+	}
+
+	for _, test := range tests {
+		script, err := txscript.NullDataScript(test.data)
+
+		if !test.valid {
+			if err == nil {
+				t.Errorf("NullDataScript test %v: expected error but none was returned. ",
+					test.name)
+			}
+
+			continue
+		}
+
+		if err != nil {
+			t.Errorf("NullDataScript test %v: Unexpected error returned %v ",
+				test.name, err)
+
+			continue
+		}
+
+		// Check that the expected result was returned.
+		if !bytes.Equal(script, test.expected) {
+			t.Errorf("NullDataScript test %v: Expected %v, got %v",
```

Comment:
You can use %x in the `Errorf` format string to get hex.

```Go
t.Errorf("%s: expected %x, got %x`, test.name, test.expected, script)
```

Also, as above, please include a `continue` after the `t.Errorf` so it continues to the next test.